gnovm/pkg/transpiler/fuzz_test.go

```diff
@@ -72,5 +240,80 @@ func FuzzTranspiling(f *testing.F) {
 			t.Fatalf("took more than 2 seconds to transpile\n\n%s", gnoSourceCode)
 		case <-doneCh:
 		}
+
+		// Next run the code to see if it can be ran.
+		fn, err := gnolang.ParseFile("main.go", string(gnoSourceCode))
+		if err != nil {
+			// TODO: it could be discrepancy that if it compiled alright that it later failed.
+			panic(err)
+		}
+
+		if !strings.Contains(gnoSrc, "func main()") {
+			gnoSrc += "\n\nfunc main() {}"
+		}
+
+		db := memdb.NewMemDB()
+		baseStore := dbadapter.StoreConstructor(db, stypes.StoreOptions{})
+		iavlStore := iavl.StoreConstructor(db, stypes.StoreOptions{})
+		store := gnolang.NewStore(nil, baseStore, iavlStore)
+		m := gnolang.NewMachine(string(fn.PkgName), store)
+		memPkg := &gnovm.MemPackage{
+			Name: string(fn.PkgName),
+			Path: string(fn.Name),
+			Files: []*gnovm.MemFile{
+				{Name: "a.gno", Body: gnoSrc},
+			},
+		}
+		if err := gnolang.TypeCheckMemPackage(memPkg, mockPackageGetter{}, false); err != nil {
```

Comment:
Comparing TypeCheckMemPackage against Go's implementation doesn't seem very useful...

Reply:
It helped rid us of bugs that you mentioned to me in issue https://github.com/gnolang/gno/issues/3729#issuecomment-2651790180 are just noisy and can help filter out the noise. It has actually been super useful for us to know what not to report.